### PR TITLE
feat: add support for working-dir in the service schema

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Closes https://github.com/canonical/rockcraft/issues/303.

It is now possible to set a working directory for Pebble services: https://github.com/canonical/pebble/commit/a807026f3b066efa00e81989900f38c2a151b73f

This PR adds support for the new field in `rockcraft.yaml`

### CODE REVIEW

File: tests/unit/test_project.py [hunk 1]
```diff
@@ -406,9 +406,10 @@ def test_full_service():
         "user-id": 1000,
         "group": "ubuntu",
         "group-id": 1000,
+        "working-dir": "/tmp",
         "on-success": "ignore",
         "on-failure": "restart",
-        "on-check_failure": {"check": "restart"},
+        "on-check-failure": {"check": "restart"},
```
Comment (SamirPS) [inline tests/unit/test_project.py:412]: How can the test pass before with the -? it's was because the ``extra=forbid`` was not here?

**Review by SamirPS — APPROVED** [review]
LGTM, just need to update this branch with main branch

File: tests/unit/test_project.py (hunk 1 above)
Comment (cjdcordeiro) [reply tests/unit/test_project.py:412]: yep

File: tests/spread/general/big/rockcraft.yaml [hunk 2]
```diff
@@ -9,6 +9,15 @@ license: Apache-2.0
 base: ubuntu:22.04
 platforms:
   amd64:
+
+services:
+  test:
+    override: replace
+    command: sh -c "pwd; sleep 5"
```
Comment (tigarmo) [inline tests/spread/general/big/rockcraft.yaml:16]: can you update `task.yaml` to check for this?

Comment (cjdcordeiro) [reply tests/spread/general/big/rockcraft.yaml:16]: oops...that was an oversight. thanks. fixed

**Review by lengau — APPROVED** [review]
This is great! Thanks!

**Review by tigarmo — APPROVED** [review]

**Pull request merged**